src/mlpack/methods/ann/layer/bilinear_interpolation.hpp

```diff
@@ -0,0 +1,124 @@
+/**
+ * @file bilinear_interpolation.hpp
+ * @author Kris Singh and Shikhar Jaiswal
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license. You should have received a copy of the
+ * 3-clause BSD license along with mlpack. If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_BILINEAR_INTERPOLATION_HPP
+#define MLPACK_METHODS_ANN_LAYER_BILINEAR_INTERPOLATION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Definition and implementation of the Bilinear Interpolation layer.
```

Comment:
Can you describe a little bit more about what the bilinear interpolation layer is and what it does in this comment please?  Keep in mind that users will read this when they are trying to figure out how and where to use this layer.  (I know that the original comment here may not have been yours, but it would be great if you could improve it regardless!)